protocol/v2/ssv/validator/non_committee_validator.go

```diff
@@ -59,19 +61,22 @@ type CommitteeObserverOptions struct {
 func NewCommitteeObserver(msgID spectypes.MessageID, opts CommitteeObserverOptions) *CommitteeObserver {
 	// TODO: does the specific operator matters?
 
-	return &CommitteeObserver{
-		msgID:                  msgID,
-		logger:                 opts.Logger,
-		Storage:                opts.Storage,
-		beaconNetwork:          opts.NetworkConfig.Beacon,
-		networkConfig:          opts.NetworkConfig,
-		ValidatorStore:         opts.ValidatorStore,
-		newDecidedHandler:      opts.NewDecidedHandler,
-		attesterRoots:          opts.AttesterRoots,
-		syncCommRoots:          opts.SyncCommRoots,
-		domainCache:            opts.DomainCache,
-		postConsensusContainer: make(map[phase0.ValidatorIndex]*ssv.PartialSigContainer),
+	co := &CommitteeObserver{
+		msgID:             msgID,
+		logger:            opts.Logger,
+		Storage:           opts.Storage,
+		beaconNetwork:     opts.NetworkConfig.Beacon,
+		networkConfig:     opts.NetworkConfig,
+		ValidatorStore:    opts.ValidatorStore,
+		newDecidedHandler: opts.NewDecidedHandler,
+		attesterRoots:     opts.AttesterRoots,
+		syncCommRoots:     opts.SyncCommRoots,
+		domainCache:       opts.DomainCache,
 	}
+
+	co.postConsensusContainer = make(map[phase0.Slot]map[phase0.ValidatorIndex]*ssv.PartialSigContainer, co.postConsensusContainerCapacity())
```

Comment:
why not inside of `&CommitteeObserver{ ... }?

Reply:
because you need to call `co.postConsensusContainerCapacity()`